### ISSUE TITLE
Error in multi-GPU LoRA fine-tuning

### ISSUE THREAD

**Issue by yang-0201** [stepfun-ai/Step1X-Edit#120]
The local environment is properly configured to support single-GPU LoRA fine-tuning for Step1X, but errors occur when attempting to use multiple GPUs.

[rank0]:[W717 13:43:26.417514575 ProcessGroupNCCL.cpp:4115] [PG ID 0 PG GUID 0 Rank 0]  using GPU 0 to perform barrier as devices used by this process are currently unknown. This can potentially cause a hang if this rank to GPU mapping is incorrect.Specify device_ids in barrier() to force use of a particular device,or call init_process_group() with a device_id.

finetuning.sh:
accelerate launch  --mixed_precision bf16 --num_cpu_threads_per_process 2 --num_processes 2 \
--config_file ./library/accelerate_config.yaml \
finetuning.py \
--pretrained_model_name_or_path /home/data/Step1X-Edit/models/step1x-edit-v1p1-official.safetensors \
--qwen2p5vl /home/data/qwen/Qwen2.5-VL/models \
--ae /home/data/Step1X-Edit/models/vae.safetensors \
--cache_latents_to_disk --save_model_as safetensors --sdpa --persistent_data_loader_workers \
--max_data_loader_n_workers 4 --seed 42 --gradient_checkpointing --mixed_precision bf16 --save_precision bf16 \
--network_module library.lora_module --network_dim 16 --network_alpha 16 --network_train_unet_only \
--optimizer_type adamw8bit --learning_rate 1e-4 \
--cache_text_encoder_outputs --cache_text_encoder_outputs_to_disk \
--highvram --max_train_epochs 100 --save_every_n_epochs 5 --dataset_config library/data_configs/step1x_edit_text_610.toml \
--output_dir output_lora \
--output_name step1x-edit_test_nan_nv \
--timestep_sampling shift --discrete_flow_shift 3.1582 --model_prediction_type raw --guidance_scale 1.0 --fp8_base

accelerate_config.yaml:
compute_environment: LOCAL_MACHINE
debug: false
distributed_type: MULTI_GPU
# distributed_type: NO

downcast_bf16: 'no'
enable_cpu_affinity: false
gpu_ids: 0,1
machine_rank: 0
main_training_function: main
mixed_precision: bf16
num_machines: 1
num_processes: 2
rdzv_backend: static
same_network: true
tpu_env: []
tpu_use_cluster: false
tpu_use_sudo: false
use_cpu: false

**Comment by tingxueronghua** [comment]
I think you can directly set the num_processes in bash scripts without changing the yaml... I am not sure what will happen when setting gpu_ids.

**Comment by yang-0201** [comment]
The issue was resolved by using PyTorch version 2.7.
